src/grpc/grpc_utils.cc

```diff
@@ -73,6 +77,18 @@ operator<<(std::ostream& out, const Steps& step)
 void
 GrpcStatusUtil::Create(::grpc::Status* status, TRITONSERVER_Error* err)
 {
+  const char* dstr = getenv("TRITONSERVER_SERVER_DELAY_GRPC_RESPONSE_SEC");
```

Comment:
Instead of capturing this artificial delay here, I would recommend placing the sleep here: https://github.com/triton-inference-server/server/blob/main/src/grpc/grpc_server.cc#L237
This will cause all non-infer response writes to be delayed by the specified time interval.
Additionally, reading an environment variable is a bit heavy and can impact performance. I would recommend to call getenv once in each handler and then reuse it every time when writing the response.